raft/src/test/java/org/apache/kafka/raft/MockLog.java

```diff
@@ -418,48 +414,49 @@ public RawSnapshotWriter createSnapshot(OffsetAndEpoch snapshotId) {
     public void onSnapshotFrozen(OffsetAndEpoch snapshotId) {}
 
     @Override
-    public boolean deleteBeforeSnapshot(OffsetAndEpoch logStartSnapshotId) {
-        if (logStartOffset() > logStartSnapshotId.offset ||
-            highWatermark.offset < logStartSnapshotId.offset) {
+    public boolean deleteBeforeSnapshot(OffsetAndEpoch snapshotId) {
+        if (logStartOffset() > snapshotId.offset ||
+            highWatermark.offset < snapshotId.offset) {
 
             throw new OffsetOutOfRangeException(
                 String.format(
                     "New log start (%s) is less than start offset (%s) or is greater than the high watermark (%s)",
```

Comment:
nit: maybe we can just have two separate cases so that we can give a clearer message

Reply:
Done.